### PR TITLE
Bug1714200 m1 arm64 native

### PR DESCRIPTION
**Pull request by davehouse**
macmini m1 apple silicon native firefox testing
* arm64 python3 3.9.6 from s3 (from python.org)
* arm64 taskcluster binaries 44.0.0 from s3 (latest release; works with services current at v43.2.0)
* telegraf (monitoring and system+taskcluster logs) -- full configuration in a later PR
  * fixed path to puppet last_run_summary.yaml

### CODE REVIEW

**Review by aerickson — APPROVED** [review]
lgtm

**Review by aerickson — DISMISSED** [review]
lgtm

File: modules/packages/manifests/telegraf.pp [hunk 1]
```diff
@@ -2,9 +2,17 @@
...
-class packages::telegraf {
+class packages::telegraf (
+    Pattern[/^\d+\.\d+\.\d+_?\d*$/] $version = '1.12.3',
+) {
+    # arm64 if Apple processor
+    if /^Apple.*/ in $facts['processors']['models'] {
+        $suffix = '-arm64'
+    } else {
+        $suffix = ''
+    }
```
Comment (dividehex) [inline modules/packages/manifests/telegraf.pp:13]: Another way to test for M1 is `model_identifier => "Macmini9,1",` if you want to keep it consistent with existing methods.  Not a big deal, just a nit.

**Review by dividehex — DISMISSED** [review]

File: modules/packages/manifests/mercurial.pp [hunk 2]
```diff
@@ -15,14 +15,10 @@
...
-    # libdir = '../../Library/Python/2.7/site-packages/'
-    # but installs them in:
-    # /Library/Python/2.7/site-packages/
-    # So, link /Library under /usr to make hg find it.
-    # Not needed for bigsur on M1
-    if ! member(['10.15', '11.0', '11.1', '11.2'], $facts['os']['macosx']['version']['major']) {
+    # libdir = '../../Library/Python/2.7/site-packages/'   # 5.1 pkg hg (LIBDIR)
+    # libdir = '../../../Library/Python/2.7/site-packages' # 5.5+ pkg hg
+    if ! '11' == $facts['os']['macosx']['version']['major']
+      and $version == '5.1' {
```
[the remark this answers is not in the document]
Comment (davehouse) [reply modules/packages/manifests/mercurial.pp:21]: Okay I'll pull in latest and add comments instead.

File: modules/packages/manifests/telegraf.pp (hunk 1 above)
Comment (davehouse) [reply modules/packages/manifests/telegraf.pp:13]: I understand matching previous checking of macmini model. For this I like that the condition shows that we are looking for the processor architecture and *could work with any apple silicon model.

Comment (dividehex) [reply modules/packages/manifests/telegraf.pp:13]: That is fine.  FYI, "Macmini9,1" models are only M1 arm64 arch.  There are no x86_64 models and there never will be.

**Pull request merged**